main.star

```diff
@@ -1,14 +1,50 @@
-input_parser = import_module("./src/package_io/input_parser.star")
 ethereum_package = import_module("github.com/kurtosis-tech/ethereum-package/main.star")
 contract_deployer = import_module("./src/contracts/contract_deployer.star")
 static_files = import_module(
     "github.com/kurtosis-tech/ethereum-package/src/static_files/static_files.star"
 )
-participant_network = import_module("./src/participant_network.star")
-blockscout = import_module("./src/blockscout/blockscout_launcher.star")
+l2_launcher = import_module("./src/l2.star")
 
+def run(plan, args={}): 
+    """Deploy Optimism L2s on an Ethereum L1.
 
-def get_l1_stuff(all_l1_participants, l1_network_params):
+    Args:
+        args(yaml): Configures other aspects of the environment.
+    Returns:
+        Full Optimism L2s. 
+    """
+    plan.print("Parsing the L1 input args")
+    ethereum_args = args["l1"]
+
+    # Deploy the L1
+    plan.print("Deploying a local L1")
+    l1 = ethereum_package.run(plan, ethereum_args)
+    
+    # Get L1 info
+    all_l1_participants = l1.all_participants
+    l1_network_params = l1.network_params
+    l2_num = 0
+    num_l2s = len(args["l2s"])
+    l1_priv_keys = []
+    for i in range(0, num_l2s):
+        if 12 + l2_num > len(l1.pre_funded_accounts) - 1:
+            fail("cannot support this many l2s yet")
+        l1_priv_key = l1.pre_funded_accounts[12 + l2_num].private_key  # reserved for L2 contract deployers
+        l1_priv_keys.append(l1_priv_key)
+        l2_num += 1
+    plan.print("l1 private keys for contract deployers {0}".format(l1_priv_keys))
```

Comment:
This is pretty hacky lmao. 

Wouldn't it make more sense to fund the deployer accounts with 1000ETH and use those funds to later fund the rest of the accounts?